File: src/main/java/org/micks/DiscGolfApplication/connection/DatabaseConfigProperties.java
package org.micks.DiscGolfApplication.connection;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "db")
@Getter
@Setter
public class DatabaseConfigProperties {

    private String name;
    private String user;
    private String password;
}

File: src/main/java/org/micks/DiscGolfApplication/connection/DatabaseConfigProperties.kt
package org.micks.DiscGolfApplication.connection

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@ConfigurationProperties(prefix = "db")
data class DatabaseConfigProperties(
    var host: String? = null,
    var name: String? = null,
    var user: String? = null,
    var password: String? = null,
)